### DISCUSSION TITLE
I cannot get the last modified date to work properly

### DISCUSSION THREAD

**Discussion by MavropaliasG** [SilentVoid13/Templater#1195] [Help] [answered]
Hi, I would like to have a template that when I create a file based on it, to have a field 'Created' assign the date and time of the creation of the file (not the template), then each time I edit the file (not the template which it was created from) have a 'Last modified' field update the last modified update the date and time.

I've created a template with the following header:

```
---
Created: <% tp.file.creation_date("YYYY-MM-DD HH:mm:ss") %>
Last modified: <% tp.file.last_modified_date("YYYY-MM-DD HH:mm:ss") %>
Due: 
Category: X
Status: Backlog
Started: false
Needs action: true
Completed: false
Priority: 0
tags:
  - research
---
```

Once I create a new note with this template, it assigns the created date to the _template_ creation date and not the _file_ creation date. Moreover, the last modified date is not updating when I update the file. What I noticed is that once the file is created using the template, the 'Created' and 'Last modified' are changed from code to text:

```
---
Created: 2023-09-16 12:08:31
Last modified: 2023-09-16 12:08:31
Due: 
Category: X
Status: Backlog
Started: false
Needs action: true
Completed: false
Priority: 0
tags:
  - research
---
``` 

How can I make them remain dynamic in the header of the file?

Also, when I tried using [Dynamic commands](https://silentvoid13.github.io/Templater/commands/dynamic-command.html) from the documentation, I entered this in the text `Last modified date: <%+ tp.file.last_modified_date() %>` which displays this `Last modified date: NaN`. Is this a bug?

**Comment by Zachatoo** [comment, marked as answer]
It is a bug, I don't recommend using dynamic commands.

#1015
#910
#913

Use the Linter plugin instead for setting created/modified dates in frontmatter.

**Reply by RayBB** [reply]
Adding context for people looking into this in the future.

Linter Plugin: https://platers.github.io/obsidian-linter/
However, you should know that uses the time the file was modified according to the filesystem, not the time it was modified according to when you edited it in obsidian so if another program edits it then the updated time will change.

If you want it to only change the updated time when you update in obsidian you should try this one:
https://obsidian.md/plugins?id=frontmatter-modified-date

Relevant Discussion: https://forum.obsidian.md/t/how-to-add-an-automatically-updated-last-edit-time-in-document-properties/67224/7
